### PR TITLE
feat: upsert schema api changes to save commit_sha and source_url

### PR DESCRIPTION
**Pull request by gdgagan696**

### CODE REVIEW

File: core/changedetector/change_detector.go [hunk 1]
```diff
@@ -10,6 +10,8 @@ type ChangeRequest struct {
 	OldData     []byte
 	NewData     []byte
 	Depth       int32
+	SourceUrl   string
```
Comment (sbchaos) [inline core/changedetector/change_detector.go]: `SourceUrl` -> `SourceURL`

Please refer https://go.dev/wiki/CodeReviewComments#initialisms

File: core/schema/schema.go [hunk 2]
```diff
@@ -12,6 +13,7 @@ type Metadata struct {
 	Authority     string
 	Format        string
 	Compatibility string
+	SourceUrl     string
```
Comment (sbchaos) [inline core/schema/schema.go]: `SourceUrl` -> `SourceURL`

File: core/schema/schema.go [hunk 3]
```diff
@@ -28,7 +30,7 @@ type SchemaFile struct {
 }
 
 type Repository interface {
-	Create(ctx context.Context, namespace string, schema string, metadata *Metadata, versionID string, schemaFile *SchemaFile) (version int32, err error)
+	Create(ctx context.Context, namespace string, schema string, metadata *Metadata, versionID string, schemaFile *SchemaFile, commitSHA string) (version int32, err error)
```
Comment (sbchaos) [inline core/schema/schema.go]: Too many parameters to the function (7), can we check if we can reduce their number.

File: core/schema/schema.go [hunk 4]
```diff
@@ -62,6 +64,7 @@ type Schema struct {
 	Format        string
 	Compatibility string
 	Authority     string
+	SourceUrl     string
```
Comment (sbchaos) [inline core/schema/schema.go]: Same as above

File: core/changedetector/change_detector.go (hunk 1 above)
Comment (gdgagan696) [reply core/changedetector/change_detector.go]: might have missed as taken commitSHA. Fixed it

File: core/schema/schema.go (hunk 2 above)
Comment (gdgagan696) [reply core/schema/schema.go]: fixed

File: core/schema/schema.go (hunk 4 above)
Comment (gdgagan696) [reply core/schema/schema.go]: fixed

File: core/schema/schema.go (hunk 3 above)
Comment (gdgagan696) [reply core/schema/schema.go]: addressed

**Review by sbchaos — APPROVED** [review]

**Pull request merged**